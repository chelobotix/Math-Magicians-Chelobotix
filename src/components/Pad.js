// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';

const CalcButton = (props) => {
  const { symbol, function1 } = props;
  return (
    <button type="button" onClick={() => function1(symbol)} className="symbol">{symbol}</button>
  );
};

CalcButton.defaultProps = { symbol: '', function1: '' };
CalcButton.propTypes = { symbol: PropTypes.string, function1: PropTypes.func };

export default CalcButton;
