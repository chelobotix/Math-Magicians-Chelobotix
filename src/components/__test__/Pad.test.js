import renderer from 'react-test-renderer';
import Pad from '../Pad';

describe('Pad Component', () => {
  test('Tree', () => {
    const tree = renderer.create(<Pad />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
