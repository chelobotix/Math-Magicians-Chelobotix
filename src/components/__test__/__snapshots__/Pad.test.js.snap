// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pad Component Tree 1`] = `
<button
  className="symbol"
  onClick={[Function]}
  type="button"
/>
`;
